ouroboros-consensus/src/unstable-consensus-testlib/Test/Ouroboros/Consensus/ChainGenerator/BitVector.hs

```diff
@@ -162,11 +166,17 @@ fillInWindow ::
   -> g
   -> C.MVector base SlotE s S
   -> ST s (C.Var base (PreImage pol ActiveSlotE))   -- ^ the count after filling
-fillInWindow pol (SomeDensityWindow k s) g mv = do
+fillInWindow pol (SomeDensityWindow k s) g mv
+        | not (C.getCount k <= C.getCount s) =
+            error $ "fillInWindow: assertion failure: k <= s: "
+                    ++ show k ++ " <= " ++ show s
+        | not (C.getCount sz <= C.getCount s) =
+            error $ "fillInWindow: assertion failure: sz <= s: "
+                    ++ show sz ++ " <= " ++ show s
+        | otherwise = do
     -- how many active polarized slots @actual@ currently has
     initialActives <- countActivesInMV pol mv
 
-    let sz = C.lengthMV mv :: C.Size base SlotE
 
     -- discount the numerator accordingly if @actual@ is smaller than @s@
```

Comment:
There are two occurences of `@actual@` (here and just below) left.

Reply:
Adopted.